context/app/static/js/components/organ/Assays/Assays.jsx

```diff
@@ -54,9 +56,14 @@ function Assays({ organTerms }) {
   );
 
   const { searchData } = useSearchData(query);
-  const buckets = searchData.aggregations
-    ? searchData.aggregations.mapped_data_types['assay_display_name.keyword'].buckets
-    : [];
+
+  useEffect(() => {
+    if (searchData && searchData.aggregations) {
+      const buckets = searchData.aggregations.mapped_data_types['assay_display_name.keyword'].buckets || [];
+      setBucketData(buckets);
+      if (buckets.length === 0) onHandleSearchStatus(false);
+    }
+  }, [searchData, onHandleSearchStatus]);
```

Comment:
Objects, like `searchData`, and arrays provided in the dependencies are compared by reference and will continuously trigger if not memoized, handled with a custom compare fn, or handled using a custom hook like `useDeepCompareEffect` which performs a deep comparison. In this case, if we wanted to track whether the `searchData` has buckets in state we could define `const hasBuckets = buckets.length` which could then be used in the `useEffect` without issue.